crates/matrix-sdk/src/sliding_sync.rs

```diff
@@ -1381,160 +1531,29 @@ impl SlidingSyncView {
-                            break;
-                        } else {
-                            // let's check the next position;
-                            dif += 1;
-                        }
-                    }
-                    rooms_list.insert_cloned(pos, room);
-                }
-                v4::SlidingOp::Invalidate => {
-                    let max_len = rooms_list.len();
-                    let (mut pos, end): (u32, u32) = if let Some(range) = op.range {
-                        (
-                            range.0.try_into().map_err(|e| {
-                                Error::BadResponse(format!("`range.0` not a valid int: {e:}"))
-                            })?,
-                            range.1.try_into().map_err(|e| {
-                                Error::BadResponse(format!("`range.1` not a valid int: {e:}"))
-                            })?,
-                        )
-                    } else {
-                        return Err(Error::BadResponse(
-                            "`range` must be given on `Invalidate` operation".to_owned(),
-                        ));
-                    };
-
-                    if pos > end {
-                        return Err(Error::BadResponse(
-                            "Invalid invalidation, end smaller than start".to_owned(),
-                        ));
-                    }
-
-                    // ranges are inclusive up to the last index. e.g. `[0, 10]`; `[0, 0]`.
-                    // ensure we pick them all up
-                    while pos <= end {
-                        if pos as usize >= max_len {
-                            break; // how does this happen?
-                        }
-                        let idx = pos as usize;
-                        let entry = if let Some(RoomListEntry::Filled(b)) = rooms_list.get(idx) {
-                            Some(b.clone())
-                        } else {
-                            None
-                        };
-
-                        if let Some(b) = entry {
-                            rooms_list.set_cloned(pos as usize, RoomListEntry::Invalidated(b));
-                        } else {
-                            rooms_list.set_cloned(pos as usize, RoomListEntry::Empty);
-                        }
-                        pos += 1;
-                    }
-                }
-                s => {
-                    warn!("Unknown operation occurred: {:?}", s);
-                }
-            }
-        }
-
-        Ok(())
-    }
-
     #[instrument(skip(self, ops))]
     fn handle_response(
         &self,
         rooms_count: u32,
         ops: &Vec<v4::SyncOp>,
         ranges: &Vec<(usize, usize)>,
     ) -> Result<bool, Error> {
+        let current_rooms_count = self.rooms_count.get();
+        if current_rooms_count.is_none()
+            || current_rooms_count == Some(0)
+            || self.is_cold.load(Ordering::Relaxed)
```

Comment:
Why `Ordering::Relaxed`?

Reply:
I think unless it's perf-relevant, one should default to `SeqCst`. It reduces the potential for very hard to debug issues.